diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -73,7 +73,9 @@ Image.gz-dtb: vmlinux scripts dtbs
 
 dtbs: scripts
 	$(Q)$(MAKE) $(build)=$(boot)/dts dtbs
-	$(Q)$(MAKE) $(build)=$(boot)/dts/qcom dtbs
+	$(Q)if [ -d $(srctree)/$(boot)/dts/qcom ]; then \
+		$(MAKE) $(build)=$(boot)/dts/qcom dtbs; \
+	fi
 
 PHONY += vdso_install
 vdso_install:
diff --git a/arch/arm64/boot/Makefile b/arch/arm64/boot/Makefile
--- a/arch/arm64/boot/Makefile
+++ b/arch/arm64/boot/Makefile
@@ -14,7 +14,7 @@
 # Based on the ia64 boot/Makefile.
 #
 
-include $(srctree)/arch/arm64/boot/dts/qcom/Makefile
+-include $(srctree)/arch/arm64/boot/dts/qcom/Makefile
 
 targets := Image Image.gz
